contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -25,33 +25,35 @@
 
 
 class GossipStoreHeader(object):
-    def __init__(self, buf: bytes):
-        length, self.crc, self.timestamp = struct.unpack('>III', buf)
-        self.deleted = (length & GOSSIP_STORE_LEN_DELETED_BIT) != 0
-        self.length = (length & GOSSIP_STORE_LEN_MASK)
+    def __init__(self, buf: bytes, off: int):
+        self.flags, self.length, self.crc, self.timestamp = struct.unpack('>HHII', buf)
+        self.off = off
+        self.deleted = (self.flags & GOSSIP_STORE_LEN_DELETED_BIT) != 0
+        self.ratelimit = (self.flags & GOSSIP_STORE_LEN_RATELIMIT_BIT) != 0
+        self.zombie = (self.flags & GOSSIP_STORE_ZOMBIE_BIT) != 0
 
 
 class GossmapHalfchannel(object):
     """One direction of a GossmapChannel."""
     def __init__(self, channel: 'GossmapChannel', direction: int,
-                 timestamp: int, cltv_expiry_delta: int,
-                 htlc_minimum_msat: int, htlc_maximum_msat: int,
-                 fee_base_msat: int, fee_proportional_millionths: int):
-
+                 fields: Dict[str, Any], hdr: GossipStoreHeader):
```

Comment:
Yes, but the `[fields](fields: Dict[str, Any])` is returned by the `pyln.spec.bolt7.channel_update.read()` anyway. (In the `_update_channel() method`.)
I simply used that as before, changing/optimizing this would require a bigger rewrite maybe done in a dedicated PR?